src/main/java/org/opensearch/security/ssl/http/netty/Netty4HttpRequestHeaderVerifier.java

```diff
@@ -94,20 +96,26 @@ public void channelRead0(ChannelHandlerContext ctx, DefaultHttpRequest msg) thro
         ThreadContext threadContext = threadPool.getThreadContext();
         try (ThreadContext.StoredContext ignore = threadPool.getThreadContext().stashContext()) {
             injectUser(msg, threadContext);
-            // If request channel is completed and a response is sent, then there was a failure during authentication
-            restFilter.checkAndAuthenticateRequest(requestChannel);
+
+            boolean shouldSkipAuthentication = HttpMethod.OPTIONS.equals(msg.method())
+                || HEALTH_SUFFIX.equals(suffix)
+                || WHO_AM_I_SUFFIX.equals(suffix);
```

Comment:
Nit: it would be nice to refactor this into another class, there are a couple places where we've got duplicate logic